FWCore/ParameterSet/python/Config.py

```diff
@@ -538,6 +538,9 @@ def extend(self,other,items=()):
                 self.add_(item)
             elif isinstance(item,ProcessModifier):
                 item.apply(self)
+            elif isinstance(item,Process):
+                fragment = ProcessWrapper(item)
+                self.extend(fragment)
```

Comment:
As for the Schedule, so far I've reverted to the original behavior to avoid
messing up cmsDriver.

I'll think about it again once I have a better understanding of cmsDriver's
internals.

Ciao,
.Andrea
On 19 Mar 2015 17:49, "Chris Jones" <notifications@github.com> wrote:

> In FWCore/ParameterSet/python/Config.py
> <https://github.com/cms-sw/cmssw/pull/8411#discussion_r26773049>:
>
> > @@ -538,6 +538,9 @@ def extend(self,other,items=()):
> >                  self.add_(item)
> >              elif isinstance(item,ProcessModifier):
> >                  item.apply(self)
> > +            elif isinstance(item,Process):
> > +                fragment = ProcessWrapper(item)
> > +                self.extend(fragment)
>
> David Lange and I had a short discussion and I think we came up with a
> solution which allows both of us to be happy.
> You'd be able to still do process.load("some file") and get your changes
> and we'd still be able to safely had 'free roaming' cms.Process
> instances.
> The idea is to add a new class type, say ExtendFromProcess, which takes
> as its only constructor a cms.Process instance. Then we'd change the code
> above to be
>
>   elif isinstance(item, ExtendFromProcess):
>          self.extend( ProcessWrapper( item.process) )
>
> Then in your file you'd just have
>
>      f1 = cms.ExtendFromProcess(fragment)
>
> You could even make the ExtendFromProcess append the items in the Schedule
> if you wanted.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/8411/files#r26773049>.
>